paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/MergeIntoAction.java

```diff
@@ -223,20 +225,10 @@ public void validate() {
         }
 
         CoreOptions.MergeEngine mergeEngine = CoreOptions.fromMap(table.options()).mergeEngine();
-        if ((matchedUpsert || notMatchedUpsert) && !mergeEngine.supportBatchUpdate()) {
+        boolean supportMergeInto = mergeEngine == DEDUPLICATE || mergeEngine == PARTIAL_UPDATE;
```

Comment:
I think if there is only 'Not Matched Insert' we shouldn't check the merge engine.

Reply:
Consistent with Spark Merge Into.